### ISSUE TITLE
`cargo asm` doesn't show any functions.

### ISSUE THREAD

**Issue by ralfbiedert** [gnzlbg/cargo-asm#23]
When I try to disassemble [faster](https://github.com/AdamNiederer/faster), a Rust SIMD library, neither `cargo asm` nor `cargo llvm-ir` produce any output.

<img width="713" alt="screen shot 2018-07-18 at 20 00 03" src="https://user-images.githubusercontent.com/301288/42899053-30647700-8ac5-11e8-9db6-3a8c5da69293.png">

On some other crates it seemed to work though. 

`rustc 1.29.0-nightly (31f1bc7b4 2018-07-15)`

**Comment by ralfbiedert** [comment]
Yes, you are probably correct. Most it not all `faster` functions are `#[inline]`. Maybe it could just report that it didn't find anything, plus what you said, e.g.:

```
No functions found. 

Hint: Functions marked with #[inline] cannot be inspected directly. To inspect a inlined function f, wrap it in a non-inlined function g.
```

(If that is how it works)

**Comment by gnzlbg** [comment]
> Update On second thought, the "wrap f in g" might not make in all situations, as the generated code could differ quite a bit.

The best you can do is create a new crate, add `faster` as a dependency, and in the new crate, call the functions from faster that you want to inspect, but without marking these as `#[inline]`.

**Comment by ralfbiedert** [comment]
Thanks for the hint! For testing purposes I also added a non-inlined method `test` to the faster crate, and now things show up:

```
<T as core::any::Any>::get_type_id
<std::panicking::begin_panic::PanicPayload<A> as core::panic::BoxMeUp>::box_me_up
<std::panicking::begin_panic::PanicPayload<A> as core::panic::BoxMeUp>::get
core::ptr::drop_in_place
faster::test
```

I also saw the message you mentioned along my previous attempts:

```
[ERROR]: could not find function at path ...

Tips:
* make sure that the function is present in the final binary (e.g. if it's a generic function, make sure that it is actually monomorphized)
* try to do a --clean build (sometimes changes are not picked up)
```

However, I didn't make the mental connection with inlining: I knew there is no actual code for `f<T>`, but an `#[inline] f()` "should clearly be there (as inlining is only a hint anyway)".

Maybe the text could be:

```
[ERROR]: could not find functions.

Tips:
* make sure that the function is present in the final binary. 
* functions not monomorphized do not have a binary representation and do not exist in assembly.
* functions marked as #[inline] will only appear embedded within non-inlined functions.
* try to do a --clean build (sometimes changes are not picked up)
```

**Comment by gnzlbg** [comment]
> Maybe the text could be:

Yeah, I agree, the error message should be better. Maybe I have time to work on this crate again this week, there are a couple of things that need fixing and implementing.
